idaes/models_extra/power_generation/unit_models/cpu.py

```diff
@@ -0,0 +1,808 @@
+                    b.inlet_mole_frac_comp[t, 'O2'],
+                    b.inlet_mole_frac_comp[t, 'H2O'],
+                    b.inlet_mole_frac_comp[t, 'N2']
+                    ))
+
+        # Vent stream: Pressure
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Pressure: vent [Pa]")
+        def vent_pressure_eq(b, t):
+            return (
+                b.vent_pressure[t] ==
+                vent_pressure_fun.f(
+                    b.inlet_flow_mol[t],
+                    b.inlet_mole_frac_comp[t, 'Ar'],
+                    b.inlet_mole_frac_comp[t, 'CO2'],
+                    b.inlet_mole_frac_comp[t, 'O2'],
+                    b.inlet_mole_frac_comp[t, 'H2O'],
+                    b.inlet_mole_frac_comp[t, 'N2']
+                    ))
+
+    def initialize(blk,
+                   outlvl=idaeslog.NOTSET,
+                   solver='ipopt',
+                   optarg={'tol': 1e-6},
+                   release_state=True):
+        '''
+        CO2 pure pyomo block initialization routine
+
+        Keyword Arguments:
+            outlvl : sets output level of initialisation routine
+
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        '''
+        iscale.calculate_scaling_factors(blk)
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        opt = pyo.SolverFactory(solver)
+        opt.options = optarg
+
+        init_log.info_low("Starting initialization...")
+
+        blk.inlet.flow_mol[0].fix()
+        blk.inlet.mole_frac_comp[0, 'Ar'].fix()
+        blk.inlet.mole_frac_comp[0, 'CO2'].fix()
+        blk.inlet.mole_frac_comp[0, 'O2'].fix()
+        blk.inlet.mole_frac_comp[0, 'H2O'].fix()
+        blk.inlet.mole_frac_comp[0, 'N2'].fix()
+
+        # solve model
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high(
+                "Initialization Step 1 {}.".format(idaeslog.condition(res))
+            )
+        init_log.info_high("Initialization Step 1 Complete.")
+
+        # release all states and activate constraint
+        if release_state:
+            blk.inlet.flow_mol[0].unfix()
+            blk.inlet.mole_frac_comp[0, 'Ar'].unfix()
+            blk.inlet.mole_frac_comp[0, 'CO2'].unfix()
+            blk.inlet.mole_frac_comp[0, 'O2'].unfix()
+            blk.inlet.mole_frac_comp[0, 'H2O'].unfix()
+            blk.inlet.mole_frac_comp[0, 'N2'].unfix()
+
+        init_log.info("Initialization Complete.")
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        for c in self.component_data_objects(Constraint):
+            iscale.set_scaling_factor(c, 1)
```

Comment:
Scaling factors for constraints should generally be set with `iscale.constraint_scaling_transform`. If you set a constraint scaling factor, it scales the Jacobian, but IPOPT still uses the unscaled constraint residual to determine constraint convergence. Transforming the constraint has the effect of scaling both the Jacobian and the constraint residual.

Also, are you setting these constraint scaling factors because all the constraints in this model are well-scaled, or to silence some script that checks whether or not objects are scaled? If it's the latter, adding in these dummy scaling factors defeats the purpose of that script.